pkg/policyreport/eraser.go

```diff
@@ -0,0 +1,199 @@
+package policyreport
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/kyverno/kyverno/api/policyreport/v1alpha2"
+	"github.com/kyverno/kyverno/pkg/client/clientset/versioned"
+	policyreportv1alpha2listers "github.com/kyverno/kyverno/pkg/client/listers/policyreport/v1alpha2"
+	"github.com/kyverno/kyverno/pkg/config"
+	"github.com/kyverno/kyverno/pkg/toggle"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type PolicyReportEraser interface {
+	CleanupReportChangeRequests(ns string) error
+	EraseResultEntries(ns *string) error
+}
+
+func NewPolicyReportEraser(
+	pclient versioned.Interface,
+	reportLister policyreportv1alpha2listers.PolicyReportLister,
+	clusterReportLister policyreportv1alpha2listers.ClusterPolicyReportLister,
+) PolicyReportEraser {
+	return &eraser{
+		pclient:             pclient,
+		reportLister:        reportLister,
+		clusterReportLister: clusterReportLister,
+		mutex:               &sync.RWMutex{},
+	}
+}
+
+type eraser struct {
+	pclient             versioned.Interface
+	reportLister        policyreportv1alpha2listers.PolicyReportLister
+	clusterReportLister policyreportv1alpha2listers.ClusterPolicyReportLister
+	mutex               *sync.RWMutex
+}
+
+// TODO: make sure both functions below can be synced separately
+func (e *eraser) CleanupReportChangeRequests(ns string) error {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+	var selector map[string]string
+	if ns != "" {
+		selector = map[string]string{ResourceLabelNamespace: ns}
+	}
+	return e.cleanupReportChangeRequests(selector)
+}
+
+func (e *eraser) EraseResultEntries(ns *string) error {
```

Comment:
any reason why this is a pointer to a string? what happens if its `nil`?

Reply:
It was like this, I tried to change it but was afraid by regressions.
I will change that too.